db/sovereign-db/src/ledger_db/rpc.rs

```diff
@@ -76,6 +85,11 @@ impl LedgerRpcProvider for LedgerDB {
         slot_ids: &[sovereign_sdk::rpc::SlotIdentifier],
         query_mode: QueryMode,
     ) -> Result<Vec<Option<Self::SlotResponse>>, anyhow::Error> {
+        anyhow::ensure!(
+            slot_ids.len() <= MAX_SLOTS_PER_REQUEST as usize,
+            "requested too many slots. Max: {}",
```

Comment:
This is minor, but I believe it is nice to have actual length in the error message:

```suggestion
            "requested too many slots: {} . Max: {}", slot_ids.len(),
```

Reply:
Updated!